### ISSUE TITLE
container is not passed to the factory when using stringified annotations

### ISSUE THREAD

**Issue by guacs** [hynek/svcs#54]
A minimal reproduction:

```python
from __future__ import annotations

from svcs import Container, Registry


def get_int() -> int:
    return 1


def get_str(container: Container) -> str:
    return str(container.get(int))


reg = Registry()

reg.register_factory(int, get_int)
reg.register_factory(str, get_str)

container = Container(reg)

assert str(container.get(int)) == container.get(str)
```

The reason this fails is because `inspect.signature` does not resolve stringified annotations by default.

**Comment by hynek** [comment]
heh I had problems writing a test for this and then I noticed that I've already been cheating:

```
if (annot := p.annotation) is Container or annot == "svcs.Container":
```

get_type_hints is a bit slow and janky (cf other issue 😬) would you be OK if I just add `"Container"` to it and call it a day?

**Comment by guacs** [comment]
I think I have a proper fix that should work for most use cases. I'll raise a PR and you can check if you like that approach?

**Comment by hynek** [comment]
fixed by #55! 🎉
